packages/next/src/client/image.tsx

```diff
@@ -157,6 +157,19 @@ type ImageElementProps = Omit<ImageProps, 'src' | 'alt' | 'loader'> & {
   setShowAltText: (b: boolean) => void
 }
 
+/**
+ * Returns only widths smaller than imageWidth and the next wider one
+ */
```

Comment:
Shouldn't it be the other way around? The srcset should contain images larger than the width prop and omit the smaller ones.

Also, can you add a test to verify the behavior works as expected?

Also the [existing tests are failing](https://github.com/vercel/next.js/actions/runs/4774850958/jobs/8488805775?pr=48729#step:6:3048)

Reply:
`srcset` is used so devices with smaller screens can download a smaller version of an image - which has a smaller size and will take less time to be downloaded. Image optimizer doesn't upscale source images, so the highest possible resolution is going to be the same as the resolution of the source image. If we would ask it to create versions 1000x1000, 2000x2000, 3000x3000... for an image that has smaller dimensions than that, those versions would be stored in the cache, but actually, all of them will have the same size as the original image.

I will create a test for it and fix failing tests some day during this week.